src/sound_tool.c

```diff
@@ -0,0 +1,1285 @@
+#include "constants/snd_misc.h"
+#include "constants/snd_system.h"
+#include "constants/snd_def.h"
+#include "sound.h"
+#include "sound_tool.h"
+#include "unk_02005D10.h"
+#include "unk_02035900.h"
+
+static u16 Snd_GetNextBGM(void);
+static void Snd_SetSubscene(u8 scene);
+static void Snd_SetSubData(u8 scene);
+static void Snd_SetFieldData(u16 no, int flag);
+static void Snd_SetFieldDataWithoutDelay(u16, u16);
+static void Snd_SetBattleData(u16 no, int flag);
+static void Snd_SetP2PData(u16 no, int unused);
+static void Snd_SetContestData(u16, int);
+static void Snd_SetContestImageData(u16, int);
+static void Snd_AddSceneData(u8);
+static void Snd_SetDemoData(u8, u16, int);
+static void Snd_ReverseBuffer(u8* p, u32 size);
+static BOOL Snd_CaptureIsActive(void);
+static NNSSndCaptureType GF_SndCaptureGetCaptureType(void);
+static void GF_SndCaptureStopReverb(int frame);
+static void Snd_SetNextWait(int frame);
+static BOOL MusicFadeOutAndPlayNext(u8 scene, u16 no, int frame, int next_wait, u8 flag, void* adrs);
+static void SetFadeCommon(u8 scene, u16 no, int frame, int next_wait, u8 flag, void* adrs);
+static void GF_SndSetAllocatableChannelForBGMPlayer(int channel);
+static void Snd_FieldPauseOrStop(void);
+static void Snd_SetGBSoundsState(u8 val);
+static u16 GBSounds_GetDSSeqNoByGBSeqNo(u16 seq_no);
+
+static u8 auto_sample_flag;
+static u32 unk4;
+ 
+typedef struct UnkMicStruct {
+    BOOL mono_flag;
+    u32 unkc;
+} UNK_MICSTRUCT;
+
+static UNK_MICSTRUCT _021D05E8; 
+
+extern MICAutoParam _021D05F8;
+
+#define SWAVE_BUFFER_SIZE 2000
+extern s8 sWaveBuffer[SWAVE_BUFFER_SIZE];
+
+extern const u8 _020F5710[8][4];
+
+extern const u16 GS_SeqPairs[136][2];
+
+void Snd_SetCntrlBGMFlag(u8 flag) {
+    *(u8 *) GF_SdatGetAttrPtr(SOUND_WORK_CTRL_BGM_FLAG) = flag;
+}
+
+u8 Snd_GetCntrlBGMFlag(void) {
+    return *(u8 *) GF_SdatGetAttrPtr(SOUND_WORK_CTRL_BGM_FLAG);
+}
+
+void Snd_SetCurrentlyPlayingBGM(u16 no) {
+    u16* cur_bgm = (u16 *) GF_SdatGetAttrPtr(SOUND_WORK_CUR_BGM_NO);
+    if (no > SEQ_GS_P_START) {
+        Snd_SetUNK_58(no);
+        *cur_bgm = GBSounds_GetDSSeqNoByGBSeqNo(no);
+    }
+    else {
```

Comment:
No newline between closing brace and `else` clause
```suggestion
    } else {
```
Other instances follow